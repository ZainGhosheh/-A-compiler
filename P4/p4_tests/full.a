a : () -> int {
    x: int = 1;
    y: int = 2;
	return x + y;
}

b : () -> int {
    x: int = 1;
    y: int = 2;
	return x - y;
}

c : () -> int {
    x: int = 1;
    y: int = 2;
	return x * y;
}

d : () -> int {
    x: int = 1;
    y: int = 2;
	return x / y;
}

e : () -> bool {
    x: int = 1;
    y: int = 2;
	return x > y;
}

f : () -> bool {
    x: int = 1;
    y: int = 2;
	return x < y;
}

g : () -> bool {
    x: int = 1;
    y: int = 2;
	return x >= y;
}

h : () -> bool {
    x: int = 1;
    y: int = 2;
	return x <= y;
}

i : () -> bool {
    x: int = 1;
    y: int = 2;
	return x == y;
}

j : () -> bool {
    x: int = 1;
    y: int = 2;
	return x != y;
}

k : () -> bool {
    x: bool = true;
    y: bool = false;
	return x and y;
}

l : () -> bool {
    x: bool = true;
    y: bool = false;
	return x or y;
}

ll : () -> bool {
    x: bool = true;
	return x;
}
lll : () -> int {
    return 1;
}

llll : (x: int, y: bool, z: int) -> int {
    return a(x);
}

main : () -> int {
    b(1, true, 2);
    a();
    return b(1, true, 2);
}
mmm: custom {
    x: int;
    y: int;
    z: int = 1;
    f : () -> void {
        a: int;
        b: int;
        while (true) {
            while (false) {
                while (true) {
                    return;
                }
            }
        }
        if (true) {
            if (false) {
            }
        }
    }
};

SomeClass: custom {

};
adfa : () -> void {
    x:int = 2;
    if (x > 1) {
        x: SomeClass;
    }
}
adsfd : () -> void {
    toconsole "Hello, World!";
    return;
}
dsfsdfdsfa : () -> int {
    if (eh?) {
        return 1;
    }
    else {
        return 2;
    }
}

asdfsd : (x: int, y: int, z: bool) -> void {
    if (x > 1) {
        x: SomeClass;
    }
}
asasdwer : () -> int {
    x: int = 0;
    fromconsole x;
    return x;
}
aqwer : () -> void {
}
global1: int;
global233: int;
aasdfdsf1121 : () -> int {
    if (1) {
        return 1;
    } else {
        return 2;
    }
}

qwerwerqq : immutable int;

bqwerewrq : () -> void {
    x: immutable int = 1;
    y: immutable SomeClass;
    return;
}

aqwerewqr : () -> void {
    x: int = 0;
    x++;
    x--;
    return;
}


mqwereewq: custom {
    x: int;
    y: int;
    z: int = 1;
    f : () -> void {
        a: int;
        b: int;
    }
};

wqweqrwqr: custom {
    x: int;
    y: int;
    z: int = 1;
};



main2 : () -> int {
    #maybe (4+1) means (3-1) otherwise 3+2;
    #maybe x means y otherwise z; TODO
}

cafsdf : custom {
    x: int;
    y: int;
    z: int = 1;
};

aafdbvcc : () -> int {
    c: cafsdf;
    c->x = 1;
    return c->x;
}
azxcvzxcvzda : () -> int {
    if (true) {
        if (false) {
            return 1;
        } else {
            return 2;
        }
    } else {
        return 3;
    }

    if (false) {
        if (true) {
            if (false) {
            }
        }
        return 4;
    }
}
asadfsghh : () -> int {
    while (true) {
        while (false) {
            while (true) {
                return 1;
            }
        }
    }
}
afhdhhhhh : () -> bool {
    x: bool = true;
	return !x;
}

bhhhdf : () -> int {
    x: int = 1;
	return -x;
}
afdfgertrt : () -> int {
    x: int = 1;
    y: int = 2;
	return x + y * 2;
}

breterrter : () -> int {
    x: int = 1;
    y: int = 2;
	return x + (y * 2);
}

certretqqq : () -> int {
    x: int = 1;
    y: int = 2;
	return (x + y * 0) * 2;
}

dqqqwerq : () -> int {
    x: int = 1;
    y: int = 2;
	return x + -y;
}

endgfhgdf : () -> int {
    x: int = 1;
    y: int = 2;
	return x + ((!y) and true);
}
main3 : () -> int {
    a: &int;
    #b: &a; TODO
}
a32423gaf : () -> int {
	return 1;
}

badsbvx : () -> void {
    x: int;
    x = 1;
    x = 2;
    n: int = 1;
	return;
}
aafgbvx : () -> void {
	x: int = 0;
    while (x == 1) {
		a = 1;
	}
}
