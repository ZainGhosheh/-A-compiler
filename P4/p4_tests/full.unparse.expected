a{()->int} : () -> int {
    x{int} : int = 1;
    y{int} : int = 2;
    return x{int} + y{int};
}
b{()->int} : () -> int {
    x{int} : int = 1;
    y{int} : int = 2;
    return x{int} - y{int};
}
c{()->int} : () -> int {
    x{int} : int = 1;
    y{int} : int = 2;
    return x{int} * y{int};
}
d{()->int} : () -> int {
    x{int} : int = 1;
    y{int} : int = 2;
    return x{int} / y{int};
}
e{()->bool} : () -> bool {
    x{int} : int = 1;
    y{int} : int = 2;
    return x{int} > y{int};
}
f{()->bool} : () -> bool {
    x{int} : int = 1;
    y{int} : int = 2;
    return x{int} < y{int};
}
g{()->bool} : () -> bool {
    x{int} : int = 1;
    y{int} : int = 2;
    return x{int} >= y{int};
}
h{()->bool} : () -> bool {
    x{int} : int = 1;
    y{int} : int = 2;
    return x{int} <= y{int};
}
i{()->bool} : () -> bool {
    x{int} : int = 1;
    y{int} : int = 2;
    return x{int} == y{int};
}
j{()->bool} : () -> bool {
    x{int} : int = 1;
    y{int} : int = 2;
    return x{int} != y{int};
}
k{()->bool} : () -> bool {
    x{bool} : bool = true;
    y{bool} : bool = false;
    return x{bool} and y{bool};
}
l{()->bool} : () -> bool {
    x{bool} : bool = true;
    y{bool} : bool = false;
    return x{bool} or y{bool};
}
ll{()->bool} : () -> bool {
    x{bool} : bool = true;
    return x{bool};
}
lll{()->int} : () -> int {
    return 1;
}
llll{(int, bool, int)->int} : (x{int} : int, y{bool} : bool, z{int} : int) -> int {
    return a{()->int}(x{int});
}
main{()->int} : () -> int {
    b{()->int}(1, true, 2);
    a{()->int}();
    return b{()->int}(1, true, 2);
}
mmm{mmm} : custom {
    x{int} : int;
    y{int} : int;
    z{int} : int = 1;
    f{()->void} : () -> void {
        a{int} : int;
        b{int} : int;
        while (true){
            while (false){
                while (true){
                    return;
                }
            }
        }
        if (true){
            if (false){
            }
        }
    }
};
SomeClass{SomeClass} : custom {
};
adfa{()->void} : () -> void {
    x{int} : int = 2;
    if (x{int} > 1){
        x{SomeClass} : SomeClass{SomeClass};
    }
}
adsfd{()->void} : () -> void {
    toconsole "Hello, World!";
    return;
}
dsfsdfdsfa{()->int} : () -> int {
    if (eh?){
        return 1;
    } else {
        return 2;
    }
}
asdfsd{(int, int, bool)->void} : (x{int} : int, y{int} : int, z{bool} : bool) -> void {
    if (x{int} > 1){
        x{SomeClass} : SomeClass{SomeClass};
    }
}
asasdwer{()->int} : () -> int {
    x{int} : int = 0;
    fromconsole x{int};
    return x{int};
}
aqwer{()->void} : () -> void {
}
global1{int} : int;
global233{int} : int;
aasdfdsf1121{()->int} : () -> int {
    if (1){
        return 1;
    } else {
        return 2;
    }
}
qwerwerqq{immutable int} : immutable int;
bqwerewrq{()->void} : () -> void {
    x{immutable int} : immutable int = 1;
    y{immutable SomeClass} : immutable SomeClass{SomeClass};
    return;
}
aqwerewqr{()->void} : () -> void {
    x{int} : int = 0;
    x{int}++;
    x{int}--;
    return;
}
mqwereewq{mqwereewq} : custom {
    x{int} : int;
    y{int} : int;
    z{int} : int = 1;
    f{()->void} : () -> void {
        a{int} : int;
        b{int} : int;
    }
};
wqweqrwqr{wqweqrwqr} : custom {
    x{int} : int;
    y{int} : int;
    z{int} : int = 1;
};
main2{()->int} : () -> int {
}
cafsdf{cafsdf} : custom {
    x{int} : int;
    y{int} : int;
    z{int} : int = 1;
};
aafdbvcc{()->int} : () -> int {
    c{cafsdf} : cafsdf{cafsdf};
    c{cafsdf}->x{int} = 1;
    return c{cafsdf}->x{int};
}
azxcvzxcvzda{()->int} : () -> int {
    if (true){
        if (false){
            return 1;
        } else {
            return 2;
        }
    } else {
        return 3;
    }
    if (false){
        if (true){
            if (false){
            }
        }
        return 4;
    }
}
asadfsghh{()->int} : () -> int {
    while (true){
        while (false){
            while (true){
                return 1;
            }
        }
    }
}
afhdhhhhh{()->bool} : () -> bool {
    x{bool} : bool = true;
    return !x{bool};
}
bhhhdf{()->int} : () -> int {
    x{int} : int = 1;
    return -x{int};
}
afdfgertrt{()->int} : () -> int {
    x{int} : int = 1;
    y{int} : int = 2;
    return x{int} + (y{int} * 2);
}
breterrter{()->int} : () -> int {
    x{int} : int = 1;
    y{int} : int = 2;
    return x{int} + (y{int} * 2);
}
certretqqq{()->int} : () -> int {
    x{int} : int = 1;
    y{int} : int = 2;
    return (x{int} + (y{int} * 0)) * 2;
}
dqqqwerq{()->int} : () -> int {
    x{int} : int = 1;
    y{int} : int = 2;
    return x{int} + (-y{int});
}
endgfhgdf{()->int} : () -> int {
    x{int} : int = 1;
    y{int} : int = 2;
    return x{int} + ((!y{int}) and true);
}
main3{()->int} : () -> int {
    a{& int} : & int;
}
a32423gaf{()->int} : () -> int {
    return 1;
}
badsbvx{()->void} : () -> void {
    x{int} : int;
    x{int} = 1;
    x{int} = 2;
    n{int} : int = 1;
    return;
}
aafgbvx{()->void} : () -> void {
    x{int} : int = 0;
    while (x{int} == 1){
        a{()->int} = 1;
    }
}
